presto-main/src/main/java/io/prestosql/testing/MaterializedResult.java

```diff
@@ -295,9 +303,8 @@ else if (type instanceof TimeWithTimeZoneType) {
             type.writeLong(blockBuilder, packTimeWithTimeZone(nanos, offsetMinutes));
         }
         else if (type instanceof TimestampType) {
-            long micros = ((SqlTimestamp) value).getEpochMicros();
-            int precision = ((TimestampType) type).getPrecision();
-            type.writeLong(blockBuilder, micros);
+            PrestoTimestampEncoder<? extends Comparable<?>> timestampEncoder = PrestoTimestampEncoderFactory.createTimestampEncoder((TimestampType) type, DateTimeZone.UTC);
+            timestampEncoder.write(toDecodedTimestamp((SqlTimestamp) value), blockBuilder);
```

Comment:
`PrestoTimestampEncoder` is more complex than needed here. Since this is test code, let's keep it less coupled with the code under test. I think this should do

```java
long micros = ((SqlTimestamp) value).getEpochMicros();
if (((TimestampType) type).getPrecision() <= TimestampType.MAX_SHORT_PRECISION) {
    type.writeLong(blockBuilder, micros);
}
else {
    type.writeObject(blockBuilder, new LongTimestamp(micros, ((SqlTimestamp) value).getPicosOfMicros()));
}
```